### PR TITLE
[w32file] do not discard sub-second resolution for access- and writetime if possible

### PR DESCRIPTION
**Pull request by lewurm**
Fixes https://github.com/mono/mono/issues/7646

### CODE REVIEW

File: mono/metadata/w32file-unix.c [hunk 1]
```diff
@@ -1500,59 +1498,52 @@ static gboolean file_setfiletime(FileHandle *filehandle,
...
-
-		if (sizeof (utbuf.actime) == 4 && ((access_ticks - 116444736000000000ULL) / 10000000) > INT_MAX) {
-			mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_IO_LAYER_FILE, "%s: attempt to set write time that is too big for a 32bits time_t",
-				   __func__);
-			mono_w32error_set_last (ERROR_INVALID_PARAMETER);
-			return(FALSE);
-		}
+#ifdef HAVE_STRUCT_TIMEVAL
+	struct timeval times [2];
+	memset (times, 0, 2 * sizeof (struct timeval));
```
Comment (luhenry) [inline mono/metadata/w32file-unix.c]: You could use `sizeof(times)` here.

File: mono/metadata/w32file-unix.c [hunk 2]
```diff
@@ -2832,31 +2823,6 @@ GetFileSize(gpointer handle, guint32 *highsize)
 	return ret;
 }
 
-gboolean
-mono_w32file_get_times(gpointer handle, FILETIME *create_time, FILETIME *access_time, FILETIME *write_time)
```
Comment (luhenry) [inline mono/metadata/w32file-unix.c]: Also remove https://github.com/lewurm/mono/blob/1f4d75e070f07db08ceb394a9d32ce527e2187b9/mono/metadata/w32file.h#L431

**Comment by lewurm** [comment]
PR doesn't work on linux, need to investigate.

**Comment by lewurm** [comment]
should be good now. @luhenry please have another look 🙂

**Comment by marek-safar** [comment]
@monojenkins backport 2018-02

**Comment by lewurm** [comment]
@monojenkins rebase

**Pull request merged**